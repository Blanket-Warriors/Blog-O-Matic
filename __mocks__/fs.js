export const mkdir = jest.fn((a, cb) => cb())
export const writeFile = jest.fn((a, b, cb) => cb())
export const existsSync = jest.fn(() => true)
export const readFileSync = jest.fn()
