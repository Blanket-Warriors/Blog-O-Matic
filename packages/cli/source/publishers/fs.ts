import { ncp } from "ncp"
import { resolve } from "path"
import { optimizeImages } from "./helpers/optimizeImages"


export async function fsPublisher(output: string) {
  const input = resolve(process.cwd(), "resources")
  await optimizeImages({ input, output })
  ncp(resolve(process.cwd(), "posts"), resolve(output, "posts"), {}, () => {
    console.log("Done.")
  })
}
