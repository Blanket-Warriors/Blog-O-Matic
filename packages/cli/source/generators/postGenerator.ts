import { get, promisify } from "@civility/utilities"
import * as fs from "fs"
import { prompt } from "inquirer"
import { safeLoad } from "js-yaml"
import { join } from "path"
import { IPost } from "../definitions"
import { postTemplate } from "../templates"
const writeFile = promisify(fs.writeFile)


export async function postGenerator() {
  console.log("\nðŸ˜³ðŸ˜³ðŸ¤–ðŸ˜³ LET'S MAKE A BLOG POST! ðŸ˜³ðŸ˜³ðŸ¤–ðŸ˜³\n")

  const config = safeLoad(fs.readFileSync("./blog.config.yml", "utf8")) || {}

  const questions = [
    {
      default: new Date().toLocaleString(undefined, {
        day: "numeric",
        month: "long",
        year: "numeric",
      }),
      message: "title:",
      name: "title",
      type: "input",
    },
    {
      default: get(config, [ "blog", "author" ]) || "",
      message: "author:",
      name: "author",
      type: "input",
    },
    {
      default: get(config, [ "blog", "location" ]) || "",
      message: "location:",
      name: "location",
      type: "input",
    },
  ]

  const { author = "", location = "", title = "" } = await prompt(questions) as IPost
  const filename = title.replace(",", "").replace(/[^a-zA-Z0-9_.@()-]/g, "-").toLowerCase()
  await writeFile(join("posts", `${filename}.md`), postTemplate({
    author, location, title,
    CREATED: new Date().getTime(),
    UPDATED: new Date().getTime(),
  }))

  console.log(`
    Congratulations! ðŸŽ‰ ðŸŽ‰ ðŸŽ‰
    You generated a blog post!
  `)
}
