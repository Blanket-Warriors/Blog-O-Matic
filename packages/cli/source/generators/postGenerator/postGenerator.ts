import { promisify } from "@civility/utilities"
import * as fs from "fs"
import { prompt } from "inquirer"
import { dump, load } from "js-yaml"
import { join } from "path"
import { IPost } from "../../definitions"
import * as q from "../questions"
const writeFile = promisify(fs.writeFile)
const now = new Date()


export async function postGenerator() {
  console.log("\nðŸ˜³ ðŸ˜³ ðŸ¤– ðŸ˜³  LET'S MAKE A BLOG POST! ðŸ˜³ ðŸ˜³ ðŸ¤– ðŸ˜³ \n")

  const config = load(fs.readFileSync("./blog.config.yml", "utf8")) || {}

  const { author = "", title = "" } = await prompt([
    q.postTitle,
     { ...q.postAuthor, default: config.author || "" },
   ]) as IPost

  const permalink = title
    .replace(",", "")
    .replace(/[^a-zA-Z0-9_.@()-]/g, "-")
    .toLowerCase()

  const filePath = join("posts", `${permalink}.md`)
  const postData = { author, permalink, title, created: now, updated: now }
  const frontmatter = `---\n${dump(postData)}---\n`
  await writeFile(filePath, frontmatter + `# ${title}\n`)

  console.log(`
    Congratulations! ðŸŽ‰ ðŸŽ‰ ðŸŽ‰
    You generated a blog post!
  `)
}
