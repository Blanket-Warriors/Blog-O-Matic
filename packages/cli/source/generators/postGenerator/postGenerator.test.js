jest.mock("fs")
jest.mock("inquirer")


import { advanceTo } from "jest-date-mock"
import { postGenerator } from "./postGenerator"
import { writeFile } from "fs"

beforeAll(() => { console.log = jest.fn() })

test("Should call writeFile", async () => {
  advanceTo(new Date(2018, 5, 27, 0, 0, 0))
  await postGenerator()
  const [ writeFile1 ] = writeFile.mock.calls


  expect(writeFile1[0]).toBe("posts/my-blog-title.md")
  expect(writeFile1[1]).toMatchSnapshot()

  expect(console.log).toBeCalledTimes(2)
  const [ welcomeText, finishedText ] = console.log.mock.calls
  expect(welcomeText[0]).toMatchSnapshot()
  expect(finishedText[0]).toMatchSnapshot()
})
