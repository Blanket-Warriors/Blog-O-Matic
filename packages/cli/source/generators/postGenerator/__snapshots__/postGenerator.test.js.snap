// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should call writeFile 1`] = `
"---
author: my-author
id: my-blog-title
permalink: posts/my-blog-title.md
title: my-blog-title
created: 2018-06-27T06:00:00.000Z
updated: 2018-06-27T06:00:00.000Z
---
# my-blog-title
"
`;

exports[`Should call writeFile 2`] = `
"
😳 😳 🤖 😳  LET'S MAKE A BLOG POST! 😳 😳 🤖 😳 
"
`;

exports[`Should call writeFile 3`] = `
"
    Congratulations! 🎉 🎉 🎉
    You generated a blog post!
  "
`;
